common/utils/common/repo.go

```diff
@@ -68,5 +68,10 @@ func buildHTTPCloneURL(domain string, repoType types.RepositoryType, path string
 
 func buildSSHCloneURL(domain string, repoType types.RepositoryType, path string) string {
 	sshDomainWithoutPrefix := strings.TrimPrefix(domain, "ssh://")
-	return fmt.Sprintf("%s:%ss/%s.git", strings.TrimSuffix(sshDomainWithoutPrefix, "/"), repoType, path)
+	hasPort := strings.Contains(sshDomainWithoutPrefix, ":")
```

Comment:
Is it possible that the SSH URL is an IPv6 address? If so, it may contain multiple ":" characters.

Reply:
Usually it is a domain name or an ipv4 address.